src/main/java/org/wso2/carbon/esb/connector/SNMPGetBulk.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.esb.connector;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.lang.StringUtils;
+import org.apache.synapse.MessageContext;
+import org.snmp4j.PDU;
+import org.snmp4j.Snmp;
+import org.snmp4j.TransportMapping;
+import org.snmp4j.event.ResponseEvent;
+import org.snmp4j.smi.VariableBinding;
+import org.snmp4j.transport.DefaultUdpTransportMapping;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+import org.wso2.carbon.connector.core.Connector;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.IOException;
+import java.util.Vector;
+
+/*
+ * Class SNMPGetBulk is to perform SNMPGetBulk operation.
+ *
+ * @since 1.0.0
+ */
+public class SNMPGetBulk extends AbstractConnector implements Connector {
+	private Snmp snmp;
+
+	/*
+	 * Connect method which is initiating the GetBulk operation.
+	 *
+	 * @param messageContext ESB message context
+	 */
+	@Override
+	public void connect(MessageContext messageContext) throws ConnectException {
+		String oids = (String) messageContext.getProperty(SNMPConstants.OIDs);
+		String maxRepetition = (String) messageContext.getProperty(SNMPConstants.MAX_REPETITION);
+		String nonRepeater = (String) messageContext.getProperty(SNMPConstants.NON_REPEATER);
+
+		try {
+			if (StringUtils.isNotEmpty(maxRepetition)) {
+				// Create TransportMapping
+				TransportMapping transport = new DefaultUdpTransportMapping();
+				// Create Snmp object for sending data to Agent
+				snmp = new Snmp(transport);
+				// Listens for response
+				transport.listen();
+				// Create the PDU object
+				PDU pdu = new PDU();
+				SNMPUtils.addOids(oids, pdu);
+				pdu.setType(PDU.GETBULK);
+
+				//set the max-repetitions
+				pdu.setMaxRepetitions(Integer.parseInt(maxRepetition));
+
+				//set the nonrepeaters
+				if (StringUtils.isNotEmpty(nonRepeater)) {
+					pdu.setNonRepeaters(Integer.parseInt(nonRepeater));
+				}
+				if (log.isDebugEnabled()) {
+					log.debug("Sending Request to the Agent.");
+				}
+				//sending data to Agent
+				ResponseEvent response = snmp.send(pdu, SNMPUtils.getTarget(messageContext));
+				// Process Agent Response
+				if (response != null) {
+					if (log.isDebugEnabled()) {
```

Comment:
Could you see how we could de-compose these into seperate functions ? it looks like a lot of nested if/else